src/utils/helpers/structure/generateReviewerChangesRequestedProgress.ts

```diff
@@ -0,0 +1,52 @@
+import { FullStructure, SectionState, Page } from '../../types'
+
+const generateReviewerChangesRequestedProgress = (newStructure: FullStructure) => {
+  newStructure?.sortedPages?.forEach(generatePageChangeRequestProgress)
+  newStructure?.sortedSections?.forEach(generateSectionReviewProgress)
+
+  addIsChangeRequest(newStructure)
+}
+
+const generateSectionReviewProgress = (section: SectionState) => {
+  section.changeRequestsProgress = getSums(Object.values(section.pages))
+}
+
+const generatePageChangeRequestProgress = (page: Page) => {
+  const totalChangeRequests = page.state.filter(
+    ({ isChangeRequest, element: { isVisible } }) => isChangeRequest && isVisible
+  )
+  const doneChangeRequests = totalChangeRequests.filter((element) => element.isChanged)
+
+  page.changeRequestsProgress = {
+    totalChangeRequests: totalChangeRequests.length,
+    doneChangeRequests: doneChangeRequests.length,
+  }
+}
+
+const addIsChangeRequest = (newStructure: FullStructure) => {
+  const isPageRequest = ({ changeRequestsProgress }: Page) =>
```

Comment:
was wondering about the naming, did you want to make it specific to the page ? or should it be isChangeRequest ?

Reply:
Hmmm I would prefer to change this so this doesn't set the isChangeRequest here. I'll make this change (was done as part of previous commit, but now removed when I reverted)